lib/active_record_view_model/update_operation.rb

```diff
@@ -357,7 +354,58 @@ def build_updates_for_collection_association(association_data, association_updat
         clear_association_cache(model, association_data.reflection)
       end
 
-      child_viewmodels = resolve_child_viewmodels(association_data, association_update_datas, previous_child_viewmodels, update_context)
+      child_datas =
+        case association_update.type
+        when CollectionUpdate::Type::Replace
+          association_update.values
+
+        when CollectionUpdate::Type::Functional
+          child_datas =
+            previous_child_viewmodels.map do |previous_child_viewmodel|
+              UpdateData.empty_update_for(previous_child_viewmodel)
+            end
+
+          association_update.values.each do |fupdate|
+            case fupdate.type
+            when FunctionalUpdate::Type::Append
+              child_datas.concat(fupdate.values)
+
+            when FunctionalUpdate::Type::Remove
+              removed_refs = Set.new(fupdate.values.map(&:viewmodel_reference))
+              child_datas.reject! { |child_data| removed_refs.include?(child_data.viewmodel_reference) }
+
+            when FunctionalUpdate::Type::Update
+              new_datas = fupdate.values.group_by(&:viewmodel_reference)
+
+              new_datas.each do |ref, updates|
+                unless updates.length == 1
```

Comment:
This check doesn't seem to be quite enough - what if the same element is present in separate updates (or delete/append). Better to check uniqueness of all `association_update.update_datas.map(&:viewmodel_reference)`?